mobile/src/screens/survey/components/preferences-step/preferences-step.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+
+import {
+  Button,
+  ScrollView,
+  SurveyCategory,
+  TextInput,
+} from '#libs/components/components';
+import { useState } from '#libs/hooks/hooks';
+
+import { styles } from './styles';
+
+const categoryNames = [
+  'Get emotional support',
+  'Reduce stress or anxiety',
+  'Improve sleep quality',
+  'Get over with depression',
+  'Improve focus',
+  'Other',
+];
+const EMPTY_ARRAY_LENGTH = 0;
+
+const PreferencesStep: React.FC = () => {
+  const [selectedCategories, setSelectedCategories] = useState<string[]>([]);
+  const [otherText, setOtherText] = useState<string>('');
```

Comment:
Prefer using useAppForm hook

Reply:
@v0ldemar01 here with "selectedCategories" im doing "Note: we will store selected items as an array of strings" and i dont use input validation in this component. Could you explain please, how can we use useAppForm here if it is mostly for input forms and validation. i guess we can use handleSubmit from useAppForm here ?)

Reply:
Actually, we need to validate otherText field) Besides useAppForm can maintain just simple input without any restrictions